nessie/jobs/verify_canvas_api_import.py

```diff
@@ -0,0 +1,101 @@
+"""
+Copyright ©2020. The Regents of the University of California (Regents). All Rights Reserved.
+
+Permission to use, copy, modify, and distribute this software and its documentation
+for educational, research, and not-for-profit purposes, without fee and without a
+signed licensing agreement, is hereby granted, provided that the above copyright
+notice, this paragraph and the following two paragraphs appear in all copies,
+modifications, and distributions.
+
+Contact The Office of Technology Licensing, UC Berkeley, 2150 Shattuck Avenue,
+Suite 510, Berkeley, CA 94720-1620, (510) 643-7201, otl@berkeley.edu,
+http://ipira.berkeley.edu/industry-info for commercial licensing opportunities.
+
+IN NO EVENT SHALL REGENTS BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT, SPECIAL,
+INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS, ARISING OUT OF
+THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF REGENTS HAS BEEN ADVISED
+OF THE POSSIBILITY OF SUCH DAMAGE.
+
+REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
+SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF ANY, PROVIDED HEREUNDER IS PROVIDED
+"AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
+ENHANCEMENTS, OR MODIFICATIONS.
+"""
+
+from flask import current_app as app
+from nessie.externals import redshift
+from nessie.jobs.background_job import BackgroundJob
+
+"""Logic for Canvas API schema verification job."""
+
+
+class VerifyCanvasApiImport(BackgroundJob):
+
+    def run(self):
+        app.logger.info('Starting Canvas API import verification job...')
+        expected_courses = self.get_expected_courses()
+        gradebook_history_result = self.verify_import('gradebook_history', self.get_imported_gradebook_history(), expected_courses)
+        grade_change_log_result = self.verify_import('grade_change_log', self.get_imported_grade_change_log(), expected_courses)
+
+        return f"""Verified Canvas API imports for {len(expected_courses)} courses.
+                {gradebook_history_result}
+                {grade_change_log_result}"""
+
+    def get_expected_courses(self):
+        schema = app.config['REDSHIFT_SCHEMA_CANVAS']
+        results = redshift.fetch(
+            f"""SELECT course.canvas_id AS course_id, count(assign.id) AS assignment_count
+            FROM {schema}.enrollment_term_dim term
+            JOIN {schema}.course_dim course ON term.id = course.enrollment_term_id
+            LEFT OUTER JOIN {schema}.assignment_dim assign ON course.id = assign.course_id
+            WHERE term.name IN ('Spring 2020')
```

Comment:
@lyttam - Hard-coded 'Spring 2020' is expected?

Reply:
@johncrossman Yes for now the term should be hard-coded, though I will make it a parameter once we expand to include more terms.